app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_results = {}
+
+    def wrapper(*args: (int, float, bool, str, bytes, tuple)) -> Any:
+        key = f"{func.__name__}: {args}"
+        if key in cache_results.keys():
```

Comment:
There is no need to indicate that iteration should happen over keys, it is a default way of iteration in dict:
```suggestion
        if key in cache_results:
```